segmentation violation code=0x1 addr=0x0 pc=0x6d8ea5d]

```console
$ cloudlist -s s3

  _______             _____     __ 
 / ___/ /__  __ _____/ / (_)__ / /_
/ /__/ / _ \/ // / _  / / (_-</ __/
\___/_/\___/\_,_/\_,_/_/_/___/\__/ 

		projectdiscovery.io

[INF] Current cloudlist version 1.0.9 (latest)
[INF] Listing assets from provider: gcp services: s3 id: staging
alpha-storage-bucket.storage.googleapis.com
beta-backups-bucket.storage.googleapis.com
gamma-static-assets.storage.googleapis.com
delta-source-uploads.storage.googleapis.com
epsilon-cloud-data.storage.googleapis.com
zeta-terraform-state.storage.googleapis.com
eta-build-artifacts.storage.googleapis.com
theta-dev-logs.storage.googleapis.com
iota-exports-bucket.storage.googleapis.com
kappa-screenshots-bucket.storage.googleapis.com
lambda-results-bucket.storage.googleapis.com
mu-template-storage.storage.googleapis.com
nu-test-bucket.storage.googleapis.com
xi-poc-bucket.storage.googleapis.com
omicron-cloud-build.storage.googleapis.com
pi-optimized-images.storage.googleapis.com
rho-cloud-backups.storage.googleapis.com
sigma-dev-bucket.storage.googleapis.com
[INF] Found 18 Hosts for gcp (staging)
[INF] Listing assets from provider: cloudflare services:  id: 
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x6d8ea5d]

goroutine 1 [running]:
github.com/projectdiscovery/cloudlist/internal/runner.(*Runner).Enumerate(0xc001826140)
	github.com/projectdiscovery/cloudlist/internal/runner/runner.go:93 +0x8dd
main.main()
	./main.go:14 +0x89
```

--- a/pkg/providers/alibaba/alibaba.go
+++ b/pkg/providers/alibaba/alibaba.go
@@ -93,9 +93,12 @@ func (p *Provider) Services() []string {
 
 // Resources returns the provider for an resource deployment source.
 func (p *Provider) Resources(ctx context.Context) (*schema.Resources, error) {
+	finalResources := schema.NewResources()
 	if p.ecsClient != nil {
 		ecsprovider := &instanceProvider{client: p.ecsClient, id: p.id}
-		return ecsprovider.GetResource(ctx)
+		if resources, err := ecsprovider.GetResource(ctx); err == nil {
+			finalResources.Merge(resources)
+		}
 	}
-	return nil, nil
+	return finalResources, nil
 }

--- a/pkg/providers/aws/aws.go
+++ b/pkg/providers/aws/aws.go
@@ -196,7 +196,7 @@ func worker(ctx context.Context, fn getResourcesFunc, ch chan<- result) {
 }
 
 func (p *Provider) Resources(ctx context.Context) (*schema.Resources, error) {
-	finalList := schema.NewResources()
+	finalResources := schema.NewResources()
 
 	var workersWaitGroup sync.WaitGroup
 	results := make(chan result)
@@ -262,7 +262,7 @@ func (p *Provider) Resources(ctx context.Context) (*schema.Resources, error) {
 		if result.err != nil {
 			continue
 		}
-		finalList.Merge(result.resources)
+		finalResources.Merge(result.resources)
 	}
-	return finalList, nil
+	return finalResources, nil
 }

--- a/pkg/providers/azure/azure.go
+++ b/pkg/providers/azure/azure.go
@@ -111,7 +111,7 @@ func (p *Provider) Services() []string {
 
 // Resources returns the provider for an resource deployment source.
 func (p *Provider) Resources(ctx context.Context) (*schema.Resources, error) {
-	resources := &schema.Resources{}
+	resources := schema.NewResources()
 
 	if p.services.Has("vm") {
 		vmp := &vmProvider{Authorizer: p.Authorizer, SubscriptionID: p.SubscriptionID, id: p.id}

--- a/pkg/providers/cloudflare/cloudflare.go
+++ b/pkg/providers/cloudflare/cloudflare.go
@@ -91,9 +91,13 @@ func (p *Provider) Services() []string {
 
 // Resources returns the provider for an resource deployment source.
 func (p *Provider) Resources(ctx context.Context) (*schema.Resources, error) {
+	finalResources := schema.NewResources()
+
 	if p.services.Has("dns") {
 		dnsProvider := &dnsProvider{id: p.id, client: p.client}
-		return dnsProvider.GetResource(ctx)
+		if resources, err := dnsProvider.GetResource(ctx); err == nil {
+			finalResources.Merge(resources)
+		}
 	}
-	return nil, nil
+	return finalResources, nil
 }

--- a/pkg/providers/consul/consul.go
+++ b/pkg/providers/consul/consul.go
@@ -121,10 +121,10 @@ const (
 
 // Resources returns the provider for an resource deployment source.
 func (p *Provider) Resources(ctx context.Context) (*schema.Resources, error) {
-	finalList := schema.NewResources()
+	finalResources := schema.NewResources()
 	provider := &resourceProvider{client: p.client, id: p.id}
 	if resources, err := provider.GetResource(ctx); err == nil {
-		finalList.Merge(resources)
+		finalResources.Merge(resources)
 	}
-	return finalList, nil
+	return finalResources, nil
 }

--- a/pkg/providers/consul/consul.go
+++ b/pkg/providers/consul/consul.go
@@ -121,6 +121,10 @@ const (
 
 // Resources returns the provider for an resource deployment source.
 func (p *Provider) Resources(ctx context.Context) (*schema.Resources, error) {
+	finalList := schema.NewResources()
 	provider := &resourceProvider{client: p.client, id: p.id}
-	return provider.GetResource(ctx)
+	if resources, err := provider.GetResource(ctx); err == nil {
+		finalList.Merge(resources)
+	}
+	return finalList, nil
 }

--- a/pkg/providers/digitalocean/digitalocean.go
+++ b/pkg/providers/digitalocean/digitalocean.go
@@ -66,7 +66,7 @@ const apiKey = "digitalocean_token"
 
 // Resources returns the provider for an resource deployment source.
 func (p *Provider) Resources(ctx context.Context) (*schema.Resources, error) {
-	finalList := schema.NewResources()
+	finalResources := schema.NewResources()
 
 	if p.services.Has("droplet") || p.services.Has("instance") {
 		instanceprovider := &instanceProvider{client: p.client, id: p.id}
@@ -74,7 +74,7 @@ func (p *Provider) Resources(ctx context.Context) (*schema.Resources, error) {
 		if err != nil {
 			return nil, err
 		}
-		finalList.Merge(instances)
+		finalResources.Merge(instances)
 	}
 
 	if p.services.Has("app") {
@@ -83,8 +83,8 @@ func (p *Provider) Resources(ctx context.Context) (*schema.Resources, error) {
 		if err != nil {
 			return nil, err
 		}
-		finalList.Merge(apps)
+		finalResources.Merge(apps)
 	}
 
-	return finalList, nil
+	return finalResources, nil
 }

--- a/pkg/providers/fastly/fastly.go
+++ b/pkg/providers/fastly/fastly.go
@@ -64,10 +64,10 @@ func (p *Provider) Services() []string {
 
 // Resources returns the provider for an resource deployment source.
 func (p *Provider) Resources(ctx context.Context) (*schema.Resources, error) {
+	finalResources := schema.NewResources()
 	serviceProvider := &serviceProvider{client: p.client, id: p.id}
-	services, err := serviceProvider.GetResource(ctx)
-	if err != nil {
-		return nil, err
+	if services, err := serviceProvider.GetResource(ctx); err == nil {
+		finalResources.Merge(services)
 	}
-	return services, nil
+	return finalResources, nil
 }

--- a/pkg/providers/gcp/gcp.go
+++ b/pkg/providers/gcp/gcp.go
@@ -145,7 +145,7 @@ func New(options schema.OptionBlock) (*Provider, error) {
 
 // Resources returns the provider for an resource deployment source.
 func (p *Provider) Resources(ctx context.Context) (*schema.Resources, error) {
-	finalList := schema.NewResources()
+	finalResources := schema.NewResources()
 
 	if p.dns != nil {
 		cloudDNSProvider := &cloudDNSProvider{dns: p.dns, id: p.id, projects: p.projects}
@@ -153,7 +153,7 @@ func (p *Provider) Resources(ctx context.Context) (*schema.Resources, error) {
 		if err != nil {
 			return nil, err
 		}
-		finalList.Merge(zones)
+		finalResources.Merge(zones)
 	}
 
 	if p.gke != nil {
@@ -162,7 +162,7 @@ func (p *Provider) Resources(ctx context.Context) (*schema.Resources, error) {
 		if err != nil {
 			gologger.Warning().Msgf("Could not get GKE resources: %s\n", err)
 		}
-		finalList.Merge(gkeData)
+		finalResources.Merge(gkeData)
 	}
 
 	if p.compute != nil {
@@ -171,7 +171,7 @@ func (p *Provider) Resources(ctx context.Context) (*schema.Resources, error) {
 		if err != nil {
 			return nil, err
 		}
-		finalList.Merge(vmData)
+		finalResources.Merge(vmData)
 	}
 
 	if p.storage != nil {
@@ -180,7 +180,7 @@ func (p *Provider) Resources(ctx context.Context) (*schema.Resources, error) {
 		if err != nil {
 			return nil, err
 		}
-		finalList.Merge(storageData)
+		finalResources.Merge(storageData)
 	}
 
 	if p.functions != nil {
@@ -189,7 +189,7 @@ func (p *Provider) Resources(ctx context.Context) (*schema.Resources, error) {
 		if err != nil {
 			return nil, err
 		}
-		finalList.Merge(functionsData)
+		finalResources.Merge(functionsData)
 	}
 
 	if p.run != nil {
@@ -198,8 +198,8 @@ func (p *Provider) Resources(ctx context.Context) (*schema.Resources, error) {
 		if err != nil {
 			return nil, err
 		}
-		finalList.Merge(cloudRunData)
+		finalResources.Merge(cloudRunData)
 	}
 
-	return finalList, nil
+	return finalResources, nil
 }

--- a/pkg/providers/heroku/heroku.go
+++ b/pkg/providers/heroku/heroku.go
@@ -71,9 +71,12 @@ func (p *Provider) Services() []string {
 
 // Resources returns the provider for an resource deployment source.
 func (p *Provider) Resources(ctx context.Context) (*schema.Resources, error) {
+	finalResources := schema.NewResources()
 	if p.services.Has("app") {
 		provider := &instanceProvider{client: p.client, id: p.id}
-		return provider.GetResource(ctx)
+		if resources, err := provider.GetResource(ctx); err == nil {
+			finalResources.Merge(resources)
+		}
 	}
-	return nil, nil
+	return finalResources, nil
 }

--- a/pkg/providers/hetzner/hetzner.go
+++ b/pkg/providers/hetzner/hetzner.go
@@ -69,9 +69,12 @@ func (p *Provider) Services() []string {
 
 // Resources returns the provider for an resource
 func (p *Provider) Resources(ctx context.Context) (*schema.Resources, error) {
+	finalResources := schema.NewResources()
 	if p.services.Has("instance") {
 		provider := &instanceProvider{client: p.client, id: p.id}
-		return provider.GetResource(ctx)
+		if resources, err := provider.GetResource(ctx); err == nil {
+			finalResources.Merge(resources)
+		}
 	}
-	return nil, nil
+	return finalResources, nil
 }

--- a/pkg/providers/linode/linode.go
+++ b/pkg/providers/linode/linode.go
@@ -79,9 +79,12 @@ func (p *Provider) Services() []string {
 
 // Resources returns the provider for an resource deployment source.
 func (p *Provider) Resources(ctx context.Context) (*schema.Resources, error) {
+	finalResources := schema.NewResources()
 	if p.services.Has("instance") {
 		provider := &instanceProvider{client: p.client, id: p.id}
-		return provider.GetResource(ctx)
+		if resources, err := provider.GetResource(ctx); err == nil {
+			finalResources.Merge(resources)
+		}
 	}
-	return nil, nil
+	return finalResources, nil
 }

--- a/pkg/providers/namecheap/namecheap.go
+++ b/pkg/providers/namecheap/namecheap.go
@@ -90,9 +90,12 @@ func (p *Provider) Services() []string {
 
 // Resources returns the provider for an resource
 func (p *Provider) Resources(ctx context.Context) (*schema.Resources, error) {
+	finalResources := schema.NewResources()
 	if p.services.Has("domain") {
 		provider := &domainProvider{client: p.client, id: p.id}
-		return provider.GetResource(ctx)
+		if resources, err := provider.GetResource(ctx); err == nil {
+			finalResources.Merge(resources)
+		}
 	}
-	return nil, nil
+	return finalResources, nil
 }

--- a/pkg/providers/nomad/nomad.go
+++ b/pkg/providers/nomad/nomad.go
@@ -108,6 +108,12 @@ const (
 
 // Resources returns the provider for an resource deployment source.
 func (p *Provider) Resources(ctx context.Context) (*schema.Resources, error) {
-	provider := &resourceProvider{client: p.client, id: p.id}
-	return provider.GetResource(ctx)
+	finalResources := schema.NewResources()
+	if p.services.Has("nomad") {
+		provider := &resourceProvider{client: p.client, id: p.id}
+		if resources, err := provider.GetResource(ctx); err == nil {
+			finalResources.Merge(resources)
+		}
+	}
+	return finalResources, nil
 }

--- a/pkg/providers/openstack/openstack.go
+++ b/pkg/providers/openstack/openstack.go
@@ -119,9 +119,12 @@ func (p *Provider) Services() []string {
 
 // Resources returns the provider for an resource
 func (p *Provider) Resources(ctx context.Context) (*schema.Resources, error) {
+	finalResources := schema.NewResources()
 	if p.services.Has("instance") {
 		provider := &instanceProvider{id: p.id, client: p.client}
-		return provider.GetResource(ctx)
+		if resources, err := provider.GetResource(ctx); err == nil {
+			finalResources.Merge(resources)
+		}
 	}
-	return nil, nil
+	return finalResources, nil
 }

--- a/pkg/providers/scaleway/scaleway.go
+++ b/pkg/providers/scaleway/scaleway.go
@@ -76,9 +76,12 @@ const apiAccessToken = "scaleway_access_token"
 
 // Resources returns the provider for an resource deployment source.
 func (p *Provider) Resources(ctx context.Context) (*schema.Resources, error) {
+	finalResources := schema.NewResources()
 	if p.services.Has("instance") {
 		provider := &instanceProvider{instanceAPI: instance.NewAPI(p.client), id: p.id}
-		return provider.GetResource(ctx)
+		if resources, err := provider.GetResource(ctx); err == nil {
+			finalResources.Merge(resources)
+		}
 	}
-	return nil, nil
+	return finalResources, nil
 }

--- a/pkg/providers/terraform/terraform.go
+++ b/pkg/providers/terraform/terraform.go
@@ -66,9 +66,12 @@ func (p *Provider) Services() []string {
 
 // Resources returns the provider for an resource deployment source.
 func (p *Provider) Resources(ctx context.Context) (*schema.Resources, error) {
+	finalResources := schema.NewResources()
 	if p.services.Has("instance") {
 		provider := &instanceProvider{path: p.path, id: p.id}
-		return provider.GetResource(ctx)
+		if resources, err := provider.GetResource(ctx); err == nil {
+			finalResources.Merge(resources)
+		}
 	}
-	return nil, nil
+	return finalResources, nil
 }

